src/pextlib1.0/tracelib.c

```diff
@@ -650,12 +647,23 @@ static void dep_check(int sock, char *path) {
         answer(sock, "#");
     }
 
-    /* check our list of dependencies */
-    for (t = depends; t && *t; t += strlen(t) + 1) {
-        if (strcmp(t, port) == 0) {
+    /* check our list of dependencies; use binary search */
+    size_t left = 0;
+    size_t right = dependsLength;
+    while (left < right) {
+        size_t index = left + (right - left) / 2;
+        int result = strcmp(depends[index], port);
+        if (result == 0) {
+            /* found the port */
             free(port);
             answer(sock, "+");
             return;
+        } else if (result < 0) {
+            /* continue search right */
+            left = index + 1;
+        } else {
+            /* continue search left */
+            right = index;
```

Comment:
Looks like this block could be replaced with POSIX bsearch(3). Untested example:
```
if (NULL != bsearch(port, depends, dependsLength, sizeof(depends[0]), (int (*)(const void*, const void*)) pointer_strcmp)) {
    free(port);
    answer(sock, "+");
    return;
}
```